src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,393 @@
+                             *     import {
+                             *         foo
+                             *     } from "./module";
+                             */
+                            const startLine = getLineOfLocalPosition(sourceFile, importList.getStart());
+                            const endLine = getLineOfLocalPosition(sourceFile, importList.getEnd());
+                            oneImportPerLine = endLine - startLine >= 2;
+                        }
+                        else {
+                            const startLine = getLineOfLocalPosition(sourceFile, importList.elements[0].getStart());
+                            const endLine = getLineOfLocalPosition(sourceFile, insertPoint);
+                            oneImportPerLine = endLine - startLine >= importList.elements.length - 1;
+                        }
+
+                        return {
+                            newText: `,${oneImportPerLine ? context.newLineCharacter : ""}${name}`,
+                            span: { start: insertPoint, length: 0 }
+                        };
+                    }
+
+                    function getCodeActionForNamespaceImport(namespacePrefix: string): CodeAction {
+                        /**
+                         * Cases:
+                         *     import * as ns from "mod"
+                         *     import default, * as ns from "mod"
+                         *     import ns = require("mod")
+                         *
+                         * Because there is no import list, we alter the reference to include the
+                         * namespace instead of altering the import declaration. For example, "foo" would
+                         * become "ns.foo"
+                         */
+                        return createCodeAction(
+                            Diagnostics.Change_0_to_1,
+                            [name, `${namespacePrefix}.${name}`],
+                            `${namespacePrefix}.`,
+                            { start: token.getStart(), length: 0 },
+                            sourceFile.fileName
+                        );
+                    }
+                }
+
+                function getCodeActionForNewImport(moduleSpecifier?: string, insertPos?: number): CodeAction {
+                    // if not specified an insert position, try to insert after any existing imports
+                    if (!insertPos) {
+                        let lastModuleSpecifierEnd = -1;
+                        for (const moduleSpecifier of sourceFile.imports) {
+                            const end = moduleSpecifier.getEnd();
+                            if (!lastModuleSpecifierEnd || end > lastModuleSpecifierEnd) {
+                                lastModuleSpecifierEnd = end;
+                            }
+                        }
+                        insertPos = lastModuleSpecifierEnd > 0 ? lastModuleSpecifierEnd + 1 : sourceFile.getStart();

```

Comment:
If there are no existing imports, it would be good to place the first prior to the first statement (i.e. after any comments), rather than the start of the file, as a file will often start with a license header.